src/components/DatasetDetailView.jsx

```diff
@@ -97,7 +97,11 @@ export class DatasetDetailView extends React.PureComponent {
             <div className={classes.header}> Last Modified: </div>
             <div className={classes.values}> {dataset.readers} </div>
             {/*TODO hook this up to SAM?!?!?*/}
-            <div>Manage Viewers</div>
+            <div>
+              {dataset && dataset.id && (
+                <ManageUsersModal datasetUUID={dataset.id} dispatch={dispatch} />
```

Comment:
We don't want this to be too tied to datasets or readers. I think we'd like to use this modal for managing studies/custodians + datasets/{custodians, readers, discoverers}. Instead of having it be connected/passing dispatch, let's have it accept callbacks for add/remove and the list of user emails as props.